src/cli.nim

```diff
@@ -110,11 +110,24 @@ proc showError*(s: string) =
   stdout.write("\n\n")
   showHelp(exitCode = 1)
 
-proc prefix(kind: CmdLineKind): string =
-  case kind
-  of cmdShortOption: "-"
-  of cmdLongOption: "--"
-  of cmdArgument, cmdEnd, cmdError: ""
+func formatOpt(kind: CmdLineKind, key: string, val = ""): string =
+  ## Returns a string that describes an option, given its `kind`, `key` and
+  ## optionally `val`. This is useful for displaying in error messages.
+  runnableExamples:
```

Comment:
I just checked the documentation and this is really nice! The docs state that these examples are compiled and tested as part of the documentation generator. Do we run the documentation generator in our CI pipeline? If not, it might be good to do so.